### PR TITLE
Cleanup of mpileup

### PR DESCRIPTION
**Pull request by nekrut**
Major cleanup of mpileup options. Removed test that were using cached data

### CODE REVIEW

File: tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml [hunk 1]
```diff
@@ -212,21 +230,17 @@
...
-                <param checked="False" falsevalue="" label="Output per-sample read depth" name="output_per_sample_read_depth" truevalue="-D" type="boolean" />
-                <param checked="False" falsevalue="" label="Output per-sample Phred-scaled strand bias P-value" name="output_per_sample_strand_bias_p_value" truevalue="-S" type="boolean" />
+                <param checked="False" falsevalue="" label="Disable read-pair overlap detection" name="ignore_overlaps" truevalue="-x" type="boolean" help="--ignore-overlaps"/>
+                <param checked="False" falsevalue="" label="Do not skip anomalous read pairs in variant calling" name="skip_anomalous_read_pairs" truevalue="-A" type="boolean" help="--count-orphans"/>
+                <param checked="False" falsevalue="" label=" Disable probabilistic realignment for the computation of base alignment quality (BAQ)" name="disable_probabilistic_realignment" truevalue="-B" type="boolean" help="--no-BAQ; BAQ is the Phred-scaled probability of a read base being misaligned. Applying this option greatly helps to reduce false SNPs caused by misalignments"/>
```
Comment (bgruening) [inline tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml]: This leading space can be removed.

File: tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml [hunk 2]
```diff
@@ -183,21 +201,21 @@
                 <conditional name="limit_by_region">
                     <param label="Select regions to call" name="limit_by_regions" type="select">
                         <option selected="True" value="no_limit">Do not limit</option>
-                        <option value="history">From an uploaded BED file</option>
-                        <option value="paste">Paste a list of regions or BED</option>
+                        <option value="history">From an uploaded BED file  (--positions)</option>
```
Comment (bgruening) [inline tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml]: One space can be removed.

File: tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml [hunk 3]
```diff
@@ -99,44 +106,55 @@
...
                     <option value="--BCF">BCF</option>
                 </param>
-                <param checked="False" falsevalue="--uncompressed" label="Compress output" name="compressed" truevalue="" type="boolean" />
-                <param label="Phred-scaled gap extension sequencing error probability" name="gap_extension_sequencing_error_probability" type="integer" value="20" />
-                <param label="Coefficient for modeling homopolymer errors." name="coefficient_for_modeling_homopolymer_errors" type="integer" value="100" />
+                <param checked="False" falsevalue="--uncompressed" label="Compress output" name="compressed" truevalue="" type="boolean" help="--incompressed; default=False"/>
+                <param name="output_tags" optional="True" type="select" multiple="True" display="checkboxes" label="Optional tags to output" help="--output-tags">
+                    <option value="DP">DP (Number of high-quality bases)</option>
+                    <option value="DPR">DRP(Number of high-quality bases for each observed allele)</option>
+                    <option value="DV">DV(Number of high-quality non-reference bases)</option>
```
Comment (bgruening) [inline tool_collections/samtools/samtools_mpileup/samtools_mpileup.xml]: Space between name and brackets?

**Comment by bgruening** [comment]
@nekrut besides the inconsistency of spaces this looks good to me! Thanks.

**Comment by nekrut** [comment]
@bgruening = fixing spaces now

**Comment by bgruening** [comment]
Thanks Anton! Anyone else? If not I will merge this in a few hours.

**Pull request merged**